File: backend/deliverydocsumv2/api/dynamicnamecolumn.py
# backend/deliverydocsumv2/api/dynamicnamecolumn.py
from rest_framework.views import APIView
from rest_framework.response import Response
from rest_framework import status
from datetime import timedelta
from dateutil.relativedelta import relativedelta
from ..models.models import DeliveryDocSummary

class DynamicNameColumnView(APIView):
    """
    Dinamik kolon başlıkları döner.
    """
    def get(self, request, *args, **kwargs):
        try:
            latest_summary = DeliveryDocSummary.objects.latest('updated_at')
            now = latest_summary.updated_at

            column_names = {
                "today": f"Bugün = {now.strftime('%d.%m.%Y')}",
                "yesterday": f"Dün = {(now - timedelta(days=1)).strftime('%d.%m.%Y')}",
                "dayBeforeYesterday": f"Önceki Gün = {(now - timedelta(days=2)).strftime('%d.%m.%Y')}",
                "threeDaysAgo": f"Bugün - 3 Gün = {(now - timedelta(days=3)).strftime('%d.%m.%Y')}",
                "fourDaysAgo": f"Bugün - 4 Gün = {(now - timedelta(days=4)).strftime('%d.%m.%Y')}",
                "thisMonth": f"Bu Ay Toplam = {now.strftime('%m.%Y')}",
                "lastMonth": f"Bu Ay - 1 Toplam = {(now - relativedelta(months=1)).strftime('%m.%Y')}",
                "thisYear": f"Yıllık Toplam = {now.strftime('%Y')}"
            }

            return Response(column_names, status=status.HTTP_200_OK)
        except DeliveryDocSummary.DoesNotExist:
            return Response({"error": "Veri bulunamadı."}, status=status.HTTP_404_NOT_FOUND)
        except Exception as e:
            return Response({"error": str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)

File: frontend/src/components/DeliveryDocSumV2/utils/DynamicNameColumn.js
// frontend/src/components/DeliveryDocSumV2/utils/DynamicNameColumn.js
import React from 'react';
import { format, subDays, startOfMonth, startOfYear, subMonths, parseISO } from 'date-fns';

const getDynamicColumnNames = (dateString) => {
  if (!dateString) return {};

  const now = parseISO(dateString);

  return {
    today: `Bugün = ${format(now, 'dd.MM.yyyy')}`,
    yesterday: `Dün = ${format(subDays(now, 1), 'dd.MM.yyyy')}`,
    dayBeforeYesterday: `Önceki Gün = ${format(subDays(now, 2), 'dd.MM.yyyy')}`,
    threeDaysAgo: `Bugün - 3 Gün = ${format(subDays(now, 3), 'dd.MM.yyyy')}`,
    fourDaysAgo: `Bugün - 4 Gün = ${format(subDays(now, 4), 'dd.MM.yyyy')}`,
    thisMonth: `Bu Ay Toplam = ${format(startOfMonth(now), 'MM.yyyy')}`,
    lastMonth: `Bu Ay - 1 Toplam = ${format(startOfMonth(subMonths(now, 1)), 'MM.yyyy')}`,
    thisYear: `Yıllık Toplam = ${format(startOfYear(now), 'yyyy')}`
  };
};

const DynamicNameColumn = ({ dateString }) => {
  const columnNames = getDynamicColumnNames(dateString);

  return (
    <div>
      <p>{columnNames.today}</p>
      <p>{columnNames.yesterday}</p>
      <p>{columnNames.dayBeforeYesterday}</p>
      <p>{columnNames.threeDaysAgo}</p>
      <p>{columnNames.fourDaysAgo}</p>
      <p>{columnNames.thisMonth}</p>
      <p>{columnNames.lastMonth}</p>
      <p>{columnNames.thisYear}</p>
    </div>
  );
};

export { getDynamicColumnNames };
export default DynamicNameColumn;

